### ISSUE TITLE
Uri.collapseDotSegments fails on some relative paths

### ISSUE THREAD

**Issue by yonran** [akka/akka-http#2507]
It seems that when `Uri` is given a relative URI with `..`s after the first `Segment` of a relative path, the `Uri` becomes incorrectly normalized.

Testing with `"com.typesafe.akka" %% "akka-http" % "10.1.8"`, the last two `should be`s fail (the actual value is in the comment):

```scala
import akka.http.scaladsl.model.Uri
import akka.http.scaladsl.model.Uri.Path
import org.scalatest.{FlatSpec, Matchers}

class UriTest extends FlatSpec with Matchers {
  "Uri" should "normalize dot dot" in {
    Uri(scheme = "file", path = Path("a/./b")) should be (Uri(scheme = "file", path = Path("a/b")))  // ok
    Uri(scheme = "file", path = Path("../c")) should be (Uri(scheme = "file", path = Path("c")))  // ok
    Uri(scheme = "file", path = Path("a/b/c/../../d")) should be (Uri(scheme = "file", path = Path("a/d")))  // ok
    Uri(scheme = "file", path = Path("/a/../b")) should be (Uri(scheme = "file", path = Path("/b")))  // ok
    Uri(scheme = "file", path = Path("a/../../c")) should be (Uri(scheme = "file", path = Path("c")))  // c/..//c
    Uri(scheme = "file", path = Path("a/../b")) should be (Uri(scheme = "file", path = Path("b")))  // b//b
  }
}
```

**Comment by jrudolph** [comment]
Interesting. Thanks for the report, @yonran (and sorry for the delay).

Would you want to open a PR to fix that?

**Comment by jrudolph** [comment]
Fixed in #3490
